graylog2-server/src/main/java/org/graylog2/shared/bindings/providers/OkHttpClientProvider.java

```diff
@@ -97,8 +105,36 @@ public void connectFailed(URI uri, SocketAddress sa, IOException ioe) {
             };
 
             clientBuilder.proxySelector(proxySelector);
+
+            if (!isNullOrEmpty(httpProxyUri.getUserInfo())) {
+                final List<String> list = Splitter.on(":")
+                        .limit(2)
+                        .omitEmptyStrings()
+                        .splitToList(httpProxyUri.getUserInfo());
+                if (list.size() == 2) {
+                    clientBuilder.proxyAuthenticator(new ProxyAuthenticator(list.get(0), list.get(1)));
+                }
+            }
         }
 
         return clientBuilder.build();
     }
+
+    public static class ProxyAuthenticator implements Authenticator {
+        private static final String PROXY_AUTHORIZATION = "Proxy-Authorization";
+        private final String credentials;
+
+        ProxyAuthenticator(String user, String password) {
+            this.credentials = Credentials.basic(requireNonNull(user), requireNonNull(password));
```

Comment:
Using `requireNonNull()` is not too useful here, in my opinion. Also see above regarding the edge cases for `Proxy-Authentication`.

If you want to keep the null-check, at least add a message which mentions the parameter which was `null`.

```java
this.credentials = Credentials.basic(requireNonNull(user, "user"), requireNonNull(password, "password"));
```